pkg/kube/inject/webhook.go

```diff
@@ -604,12 +617,30 @@ func createPatch(pod *corev1.Pod, prevStatus *SidecarInjectionStatus, revision s
 		model.IstioCanonicalServiceRevisionLabelName: canonicalRev})...)
 
 	if rewrite {
-		patch = append(patch, createProbeRewritePatch(pod.Annotations, &pod.Spec, sic, statusPort)...)
+		patch = append(patch, createProbeRewritePatch(pod.Annotations, &pod.Spec, sic, mesh.GetDefaultConfig().GetStatusPort())...)
 	}
 
 	return json.Marshal(patch)
 }
 
+func enablePrometheusMerge(mesh *meshconfig.MeshConfig, anno map[string]string) bool {
+	// If annotation is present, we look there first
+	if val, f := anno[annotation.PrometheusMergeMetrics.Name]; f {
+		bval, err := strconv.ParseBool(val)
+		if err != nil {
```

Comment:
nit: do we want to monitor these failures with a metric for the injector ?

Reply:
turns out we already have a `validateAnnotations` function. Let me add it there. In this function we are failing hard rather than silently containing which seems reasonable. maybe we can plug this into `istioctl analyze` as well

Reply:
analyzer: https://github.com/istio/istio/pull/22826